core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -680,4 +681,31 @@ class LogManagerTest {
     time.sleep(logConfig.fileDeleteDelayMs + 1)
     verifyMetrics(1)
   }
+
+  @Test
+  def testWaitForAllToComplete(): Unit = {
+    var invokedCount = 0
+    val success: Future[Boolean] = Mockito.mock(classOf[Future[Boolean]])
+    Mockito.when(success.get()).thenAnswer(_ => invokedCount += 1)
```

Comment:
Should we not return `true` in this answer?

Reply:
```waitForAllToComplete``` does not use the return value so it does not cause casting error even though we don't return ```true```. However, it does look like a bug so I will add the return value.